### PR TITLE
Replace application diagram with a mermaid version

### PR DESCRIPTION
**Pull request by jarstelfox**
I took a few liberties in changing the nodes to better work with Mermaid's auto layout.

qa_req 0

CR 
---

I recommend looking at the rich diff

### CODE REVIEW

**Comment by djmetzle** [comment]
@jarstelfox it looks like you did a faithful job of porting the existing Excalidraw diagram to mermaid. We could declutter that a bit though, and give a slightly more concise view of the routing paths.

I poked around and came up with a starting point:
https://mermaid.live/edit#pako:eNqFVF1v2jAU_SuWHycKBTbaomkTbamE1LEKtE3T0gcT3yTejJ059goC_nuvHQLhYxtP9sk5595zbbyiseZA-zQ1LM_I4yRShJAvBRhycfGB3Br9sl2v76QAZckEfjso7JqMijEs7BNLIWj22-9QBPGQpzAG-6LNryPGWAfCo2b8lkmmYjCBUZV7778OZKqlYFNgJs4GT6NTxjTTuUiWU6sNJEYrW7EKNyvzDL5NA4C_O6kdf_CsChEPYiHsSCWGFda42DpTRgHFD21OtVwYiK3QqhrZQb5VRO8hYU5a8tmIVKiPEd2c0sIQ6lP5EVEcXkSf_8tFGpLHuuKetPwVTAzyX-3Wjget_I5Up0WfQzlfrPkzFCpXf602XFgwiskJFNphYa-pMIKnUs90rpdzJ4kWW5jscW9W8zq-I6jZQqTE6vyTrs9dAHQoYXJ0MfZFz0eoX2ZvkngTOdvOco1GLBfNEm7Ger7GVB46pw-KNwdklue7Ye11IRJt0DmYORMc_8crj0fUZjDHnvu4VOBQICMaqQ1SmbN6ulQx7WM0aFCXc2bhXjCcyZz2EyaLHTrkAie_AyW2CLhdUbvMw6MhCouWsVaJSD3ujEQ4szYv-q2W_9xMhc3czIdoFYJnzNjsz02v1ev0rlmnC72rLnvX7fJ41r65Tjpv2wm_umx3GN1sGhRC_U_lCxUeqs0rT72N6w

File: README.md [hunk 1]
```diff
@@ -100,7 +100,52 @@ The `frontend` directory is structured as follows:
...
+      LoadBalancer["Load Balancer"] -->|strapi.ifixit.com| Strapi
+      LoadBalancer -->|ifixit.com| PHPApp
+      Strapi --> RDS
+   end
+
+   subgraph PHPApp["PHP App"]
+      direction TB
+      Web
+      API
+   end
```
Comment (djmetzle) [inline README.md:146]:
```suggestion
graph LR
   User --> Browser -->|Client Request| IsNextPage
   IsNextPageYes --> EdgeNetwork
   IsNextPageNo --> LoadBalancer
   Browser <--> AlgoliaSearchAPI
   Browser <--> ShopifyStorefrontAPI
   subgraph AWS
      CloudFront
      iFixitInfrastructure
   end
   subgraph CloudFront
      direction LR
      IsNextPage{"Default Origin?"}
      IsNextPage --> IsNextPageNo["Yes"]
      IsNextPage --> IsNextPageYes["No"]
   end
   subgraph Vercel
      direction LR
      EdgeNetwork["Edge Network"] --> Next.js["Next.js"]
   end
   subgraph ExternalResources["External APIs"]
      direction LR
      ShopifyStorefrontAPI["Shopify Storefront API"]
      AlgoliaSearchAPI["Algolia Search API"]
   end
   subgraph iFixitInfrastructure["iFixit Infrastructure"]
        direction LR
      LoadBalancer["ifixitlb"] -->|strapi.ifixit.com| Strapi
      LoadBalancer -->|*.ifixit.com| app
      Strapi
   end
```

**Comment by andyg0808** [comment]
Opinion: I think the existing diagram is more readable, even though it's harder to edit. In the Mermaid version all the nodes look the same, and the layout isn't particularly helpful to understanding what's going on, in my opinion.

I totally grant that the existing diagram will be harder to update; I just think that it probably does a better job of communicating to a reader, which to me seems like the most important part here.

**Comment by djmetzle** [comment]
@andyg0808 that's a reasonable callout.

**Comment by dhmacs** [comment]
For context, when I made the diagram I considered using mermaid so that it'd be easy to update, but I was not satisfied with how the Mermaid diagrams looked like. The excalidraw file is under version control as well ([app_architecture_diagram.excalidraw](https://github.com/iFixit/react-commerce/blob/diagram-replace-with-mermaid/app_architecture_diagram.excalidraw))

**Comment by jarstelfox** [comment]
Fair enough, I'll close for now. If we want to swap to mermaid at least we will have these commits sitting around as a basis.

**Pull request closed without merging**